Source/LogFilter.swift

```diff
@@ -0,0 +1,61 @@
+//
+//  LogModifier.swift
+//
+//  Copyright (c) 2015-present Nike, Inc. (https://www.nike.com)
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import Foundation
+
+/// The ``LogFilter`` protocol defines the requirements to add your own dynamic filters to allow your own code to
+/// decide to filter out (i.e. not log) certain types of log messages based on criteria you define.
+///
+/// The canonical default implementation of this is the ``LogLevelFilter``, which will filter out messages that do
+/// not meet the minimum specified log level.
```

Comment:
I'm not sure I follow this part of the comment -- it looks to me like `Logger` has specific logic, separate from it's filters, to determine whether logs meet the minimum log level.

Reply:
Ah good callout. My original intent was to implement log level filtering with this system, but it didn't pan out, because we filter out by level _before_ we expand the autoclosure for the message, which would defeat the purpose of having autoclosure in the first place. I removed this part of the comment.